### ISSUE TITLE
Versions 1.3.1 Now expects the shell provisioner to be a PS1 file?

### ISSUE THREAD

**Issue by ferventcoder** [WinRb/vagrant-windows#145]
Why did this happen and is there a way I can set it back to CMD.exe?

This is the setting in the vagrantfile:

``` ruby
  config.vm.provision :shell, :path => "../shared/shell/main.cmd"
```

In version 1.2.3 this works appropriately, but on upgrade to 1.3.1 and it now is trying to treat it like a powershell script.

And now it errors:

```
[default] Running provisioner: shell...
#< CLIXML
<Objs Version="1.1.0.1" xmlns="http://schemas.microsoft.com/powershell/2004/04">
<S S="Error">The splatting operator '@' cannot be used to reference variables in an 
expressi_x000D__x000A_</S><S S="Error">on. '@powershell' can be used only as an 
argument to a command. To reference va_x000D__x000A_</S><S S="Error">riables 
in an expression use '$powershell'._x000D__x000A_</S><S S="Error">At 
C:\tmp\vagrant-shell.ps1:2 char:12_x000D__x000A_</S><S S="Error">+ @powershell 
&lt;&lt;&lt;&lt;  -NoProfile -ExecutionPolicy Bypass -File 
"c:\vagrantshared\_x000D__x000A_</S><S 
S="Error">shell\InstallNet4.ps1"_x000D__x000A_</S><S S="Error">    + CategoryInfo          
: ParserError: (powershell:String) [], ParseExcept _x000D__x000A_</S><S S="Error">   
ion_x000D__x000A_</S><S S="Error">    + FullyQualifiedErrorId : 
SplattingNotPermitted_x000D__x000A_</S><S S="Error"> _x000D__x000A_</S>
</Objs>
```
### Update

I tested this on 1.3.0 and it works like it should. So this was a recent change with 1.3.1.

**Comment by ferventcoder** [comment]
And what went into the latest versions to make the gem 25.6MB (from 170K)? 

http://rubygems.org/gems/vagrant-windows/versions for reference....

**Comment by sneal** [comment]
25MB?! That shouldn't be. The release target must have pulled in some extraneous files from my working directory, like the puppet msi.

**Comment by ferventcoder** [comment]
@sneal as far as the main issue?

**Comment by sneal** [comment]
Fixed in 1.3.2

Thanks for reporting this. Basically the shell provisioner stopped respecting file extensions. I took the code from core Vagrant to add support for remote scripts, but it writes everything to a temp file - which in vagrant-windows we just assume is a PowerShell script.

I reverted the inline and non-download script code to 1.3.0 behavior and added support for remote files to correctly use the source file extension.
